Bazelisk inc pipeline:  Print projects that need migration

meteorcloudy opened the pull request:
To address: https://github.com/bazelbuild/continuous-integration/issues/555

Example look: https://buildkite.com/bazel/pcloudy-test/builds/19#_

@laurentlb

fweikert commented on buildkite/aggregate_incompatible_flags_test_result.py:
```diff
@@ -56,39 +55,73 @@ def process_build_log(failed_jobs_per_flag, already_failing_jobs, log, job):
+def print_projects_need_to_migrate(failed_jobs_per_flag):
+    info_text = ["#### The following projects need migration"]
+    jobs_need_migration = set()
+    for _, jobs in failed_jobs_per_flag.items():
```
Nit: simply
for jobs in failed_jobs_per_flag.values():

meteorcloudy replied:
Thanks!

fweikert commented on buildkite/aggregate_incompatible_flags_test_result.py:
```diff
@@ -56,39 +55,73 @@ def process_build_log(failed_jobs_per_flag, already_failing_jobs, log, job):
+        link_text = get_html_link_text(name, web_url)
+        info_text.append(f"    <li>{link_text}</li>")
+    info_text.append("</ul></details>")
+    if len(info_text) == 3:
```
I'm not a big fan of checking the list length against some constant to detect a no-op. What about changing lines 101-110 to:

```
entries = ["    <li>{}</li>".format(get_html_link_text(name, web_url)) for name, web_url in job_list]
if not entries:
    return

s = "" if job_num == 1 else "s"
info_text.append(
    f"<details><summary>{job_num} job{s} need migration, click to see details</summary><ul>"
)
info_text += entries
info_text.append("</ul></details>")
```

meteorcloudy replied:
Indeed, it's better! Thanks!

meteorcloudy commented:
@fweikert I rebased and also changed the new warning section to print projects in one line.
Example: https://buildkite.com/bazel/pcloudy-test/builds/28#d0e0f8b9-2a6a-4b40-862a-78069b31a9d7
Please take another look

fweikert commented:
Nice! LGTM

fweikert commented on buildkite/aggregate_incompatible_flags_test_result.py:
```diff
@@ -56,62 +55,96 @@ def process_build_log(failed_jobs_per_flag, already_failing_jobs, log, job):
+
+    s = "" if job_num == 1 else "s"
+    info_text.append(
+        f"<details><summary>{job_num} job{s} need migration, click to see details</summary><ul>"
```
One thing: I would rather display the number of projects than the number of jobs since that seems to be a more useful metric here.

meteorcloudy replied:
Done!
